core/commands/storage.go

```diff
@@ -257,9 +289,39 @@ var storageUploadRequestChallengeCmd = &cmds.Command{
 		return nil
 	},
 	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ssID := req.Arguments[0]
+		chunkHash := req.Arguments[1]
+		challenge := make(map[string]interface{})
+		if session.SessionMap[ssID] == nil {
+			return fmt.Errorf("session id doesn't exist")
+		}
+
+		n, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+		api, err := cmdenv.GetApi(env, req)
+		if err != nil {
+			return err
+		}
+		cid, err := cid2.Parse(chunkHash)
+		if err != nil {
+			return err
+		}
+		sch, err := NewStorageChallenge(req.Context, n, api, cid)
+		if err != nil {
+			return err
+		}
+		if err = sch.GenChallenge(); err != nil {
+			return err
+		}
+		challenge[sch.ID] = sch.Hash
+		session.SessionMap[ssID].Challenge = challenge
+		session.SessionMap[ssID].Time = time.Now()
```

Comment:
Must be thread safe, any update operation should go into the session struct helper